Initializing t0 in the dependencies list.

vytjan opened the pull request:
Adding the Tier-0 as another WMAgent system to the dependencies list.

vytjan committed:
Initializing t0 in the dependencies list.

amaltaro commented:
@vytjan as we discussed today, I'm probably cutting a new release tonight. If you want this one in, please a) fix the conflicts and b) test it once again. Thanks

vytjan commented:
@amaltaro Alan, this is tested and a new T0 release can be built this way as a WMCore subsystem.
Or is there something missing?

amaltaro commented on setup_dependencies.py:
```diff
@@ -170,5 +170,36 @@
+                    'WMCore.WMBS.File'],
+        'systems': ['wmc-web', 'wmc-database', 'wmc-runtime', 'workqueue'],
+        'statics': ['src/javascript/WMCore/WebTools/Agent',
+                    'src/javascript/WMCore/WebTools/WMBS',
```
These two "WMCore/WebTools/..." is something that I need to test.
Unless I'm mistaken, this corresponds to the old WMBS service monitoring code that was in place for the agents, given that we have deprecated/removed all that - python - code I think we no longer need it. But again, it has to be tested.

vytjan replied:
Removed all 3 WebTools statics, testing it in a replay now. If the replay is fine, then we should be good with these changes.

vytjan replied:
A replay testing these went fine and I didn't spot any issues (related to WebTools or any other).
It should be safe to merge this now.

amaltaro reviewed (changes requested):
Vytas, I made some comments in the code. If you don't want to test those dependency changes, no worries, just do the rebase and that's all.

vytjan commented on bin/wmagent-mod-config:
```diff
@@ -130,6 +130,8 @@ def modifyConfiguration(config, **args):
         config.General.central_logdb_url = args["central_logdb_url"]
         config.General.centralWMStatsURL = args["wmstats_url"]
+        # Both PhEDExInjector and WorkQueueManager use it, so we better make it general
+        config.General.rucioAccount = args["rucio_account"]
```
MIssed that this is already in master, thanks! Rebased.

vytjan commented on setup_dependencies.py:
```diff
@@ -170,5 +170,36 @@
+                     'WMCore.WMBS+', 'WMCore.ResourceControl+',
+                     'WMCore.DataStructs+', 'WMCore.ReqMgr+',
+                     'Controllers+', 'WMQuality.Emulators+',
+                     'WMCore.ACDC', 'Utils'],
```
Probably T0 does not use it, it's a leftover from the copied wmagent setup.

vytjan commented:
I mIssed that the rucio config was already in master, thanks! Rebased it.
Removed ACDC and all 3 WebTools statics.
A replay testing these changes went fine and I didn't spot any issues (related to WebTools or any other).
It should be safe to merge this now.

amaltaro commented:
Thanks, Vytas! Welcome to a new era of T0 deployments ;)
